### PR TITLE
B2 141

### PR DESCRIPTION
**Pull request by pchandra19**
Usage:
helm install mayastor . -n mayastor --set "mayastor.envcontext=iova-mode=pa"

Tested on specified hardware twice and did not face any issues.

### CODE REVIEW

File: chart/templates/mayastor/io/io-engine-daemonset.yaml [hunk 1]
```diff
@@ -84,6 +84,7 @@ spec:
         - "--ptpl-dir=/var/local/io-engine/ptpl/"{{ end }}
         - "--api-versions={{ .Values.mayastor.api }}"{{ if .Values.mayastor.target.nvmf.iface }}
         - "-T={{ .Values.mayastor.target.nvmf.iface }}"{{ end }}
+        - "--env-context=--{{ .Values.mayastor.envcontext }}"
```
Comment (tiagolobocastro) [inline chart/templates/mayastor/io/io-engine-daemonset.yaml:87]:
```suggestion
        - "-T={{ .Values.mayastor.target.nvmf.iface }}"{{ end }}{{ if  .Values.mayastor.envcontext }}
        - "--env-context=--{{ .Values.mayastor.envcontext }}"{{ end }}
```

File: chart/values.yaml [hunk 2]
```diff
@@ -234,6 +234,8 @@ mayastor:
       iface: ""
       # Reservations Persist Through Power Loss State
       ptpl: true
+  # Workaorund for mayastor io-engine on Intel Xeon E-2278G CPU @ 3.40GHz
+  envcontext: ""
```
Comment (tiagolobocastro) [inline chart/values.yaml:238]: I would probably be a bit specific, example:
```suggestion
  # Pass additional arguments to the Environment Abstraction Layer.
  # Example: workaround for running on Intel Xeon E-2278G CPU @ 3.40GHz: "iova-mode=pa"
  envcontext: ""
```

**Review by tiagolobocastro — COMMENTED** [review]
Did you try the suggestion on the last comment from https://github.com/openebs/mayastor/issues/1133 ? That seems to have resolved the issue without having to modify the iova.

File: chart/templates/mayastor/io/io-engine-daemonset.yaml (hunk 1 above)
Comment (pchandra19) [reply chart/templates/mayastor/io/io-engine-daemonset.yaml]: I have added support for workaround mentioned by user. I think on Mayastor 2.0 we will not hit this issue as we bail out if the hugepage < 1024, but either ways as discussed I have included both the workarounds and implemented review comments as suggested.

File: chart/templates/mayastor/io/io-engine-daemonset.yaml [hunk 3]
```diff
@@ -83,7 +83,8 @@ spec:
         - "-p={{ .Release.Name }}-etcd:{{ .Values.etcd.service.port }}"{{ if .Values.mayastor.target.nvmf.ptpl }}
         - "--ptpl-dir=/var/local/io-engine/ptpl/"{{ end }}
         - "--api-versions={{ .Values.mayastor.api }}"{{ if .Values.mayastor.target.nvmf.iface }}
-        - "-T={{ .Values.mayastor.target.nvmf.iface }}"{{ end }}
+        - "-T={{ .Values.mayastor.target.nvmf.iface }}"{{ end }}{{ if .Values.mayastor.target.nvmf.iface }}
```
Comment (tiagolobocastro) [inline chart/templates/mayastor/io/io-engine-daemonset.yaml:86]: Think there is a typo here on the if, should be `.Values.mayastor.envcontext` ?

File: chart/values.yaml [hunk 4]
```diff
@@ -234,6 +234,10 @@ mayastor:
       iface: ""
       # Reservations Persist Through Power Loss State
       ptpl: true
+  # Pass additional arguments to the Environment Abstraction Layer.
+  # as a workaorund for mayastor io-engine on Intel Xeon E-2278G CPU @ 3.40GHz 
+  # Example: --set {proudct}.envcontext=iova-mode=pa
```
Comment (tiagolobocastro) [inline chart/values.yaml:239]: product

**Review by tiagolobocastro — APPROVED** [review]

**Comment by pchandra19** [comment]
bors try

**Comment by pchandra19** [comment]
bors merge

**Review by avishnu — APPROVED** [review]
lgtm

**Comment by pchandra19** [comment]
bors merge

**Pull request merged**